arbnode/batch_poster.go

```diff
@@ -1947,3 +2062,18 @@ func (b *BoolRing) All(value bool) bool {
 	}
 	return true
 }
+
+func recoverAddressFromSigner(signer signature.DataSignerFunc) (common.Address, error) {
```

Comment:
can we add this to a utils file?

Reply:
https://github.com/EspressoSystems/nitro-espresso-integration/pull/560/commits/298d1b838a8279871fdce03f24f019ca47b7e3d8